### PR TITLE
chore: Improve error message in case of plugin timeout

### PR DESCRIPTION
**Pull request by priteshbandi**
When a plugin exceeds the specified timeout or deadline for content processing, the current error message displayed is ```signal: killed```. 
This PR updates the error message to a more informative message: ```[plugin_name] [command_name] command execution timeout: signal: killed```

### CODE REVIEW

File: plugin/plugin.go [hunk 1]
```diff
@@ -222,6 +222,9 @@ func (c execCommander) Output(ctx context.Context, name string, command plugin.C
 	cmd.Stdout = &stdout
 	err := cmd.Run()
 	if err != nil {
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			return nil, stderr.Bytes(), fmt.Errorf("'%s %s' command execution timeout: %s", name, string(command), err.Error());
```
[the remark this answers is not in the document]
Comment (priteshbandi) [reply plugin/plugin.go:226]: 1. Not sure how notation is emitting this error message but if someone like Ratify is calling notation-go then they would get error message from line 226. Also L#176 is debug log  which is only emitted if debugging is one while this error should be emitted in most of the cases.
2. I always like to include the underline error message in libraries to help debug the isssue. Callers can filter out message as they want.
3. Timesouts for notation-go are set by caller. AFAIR notation doesnt has any timeout



1. Not sure how notation is emitting this error message but tool like Ratify who are utilizing notation-go would encounter this error message originating from line 226. Also, It's worth noting that line 176 contains a debug log, which is only displayed when debugging is enabled, whereas this error should be visible in most scenarios.
2. I usually include the underlying error message in libraries (notation-go) to facilitate troubleshooting. This allows callers to filter or process the message as needed.
3. The timeouts for notation-go are configured by the caller. To the best of my knowledge, notation itself doesn't implement any timeout functionality.

Comment (Two-Hearts) [inline plugin/plugin.go:226]:
```suggestion
			return nil, stderr.Bytes(), fmt.Errorf("'%s %s' command execution timeout: %w", name, string(command), err);
```

File: plugin/plugin_test.go [hunk 2]
```diff
@@ -203,6 +205,26 @@ func TestNewCLIPlugin_PathError(t *testing.T) {
...
+	t.Run("plugin timeout error", func(t *testing.T) {
+		if runtime.GOOS == "windows" {
+			t.Skip("skipping test on Windows")
+		}
+		expectedErrMsg := "'sleep 10' command execution timeout: signal: killed"
+		ctxWithTimout, cancel := context.WithTimeout(ctx, 10 * time.Millisecond)
+		defer cancel()
+
+		var twoSeconds proto.Command
+		twoSeconds = "10"
```
Comment (Two-Hearts) [inline plugin/plugin_test.go:218]: nit:
not sure if this should be `tenSeconds`? as the context has a 10 sec timeout above.

Comment (priteshbandi) [reply plugin/plugin_test.go]: So context has 10 millis timeout and the `sleep 2` has 2 second timeout to test the shellout timeout usecase.

**Review by Two-Hearts — COMMENTED** [review]
LGTM after fixing the CI.

**Comment by Two-Hearts** [comment]
@priteshbandi The PR needs a rebase.

File: plugin/plugin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgnote) [reply plugin/plugin.go]: 1. Can you run a quick test on Notation to find out what the final error is going to be. I just don't want users to see stuttered error messages.
2. Agree. Helpful to include underlying causes. However, in this case, `signal: killed` can be misinterpreted as `SIGKILL` being issued by the user. Given that notation-go is killing the process due to timeout and not the user, it is important to indicate so. Do you want to move `signal: killed` to a debug log?
3. Got it.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: plugin/plugin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (priteshbandi) [reply plugin/plugin.go]: 1. There is no timeout for Notation CLI so Notation users shouldn't see this error. 
2. For library users I would like to keep it in error as it shows the underlying message. Also for golang users(users of notation-go) timeout message along with `signal: killed` should not be confusing. 
3. Its an error message returned by go library and consumer of this library should understand golang conventions. Also, as per golang convention you want to surface the internal error message (like java nested exceptions).

**Review by rgnote — APPROVED** [review]
LGTM

**Pull request merged**